portfolio/src/main/java/com/google/sps/servlets/favBirdDataServlet.java

```diff
@@ -29,21 +35,42 @@
 @WebServlet("/favorite-bird")
 public class favBirdDataServlet extends HttpServlet {
 
-  private Map<String, Integer> birdVotes = new HashMap<>();
-
+  /**
+  * Fetches data with a query, sorts data into a hash map, 
+  * and translates to JSON to load to the graph
+  */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      
+    //create and prepare a query
+    Query query = new Query("Bird");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    // store each vote in a hash map
+    Map<String, Integer> birdVotes = new HashMap<>();
+    for (Entity entity : results.asIterable()) {
+        String bird = (String) entity.getProperty("birdName");
+        int currentVotes = birdVotes.containsKey(bird) ? birdVotes.get(bird) : 0;
+        birdVotes.put(bird, currentVotes + 1);
+    }
+
+    // translate to json
     response.setContentType("application/json");
     Gson gson = new Gson();
     String json = gson.toJson(birdVotes);
     response.getWriter().println(json);
   }
 
+  /** Stores bird votes in Datastore using entities */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String bird = request.getParameter("bird");
-    int currentVotes = birdVotes.containsKey(bird) ? birdVotes.get(bird) : 0;
-    birdVotes.put(bird, currentVotes + 1);
+
+    Entity birdEntity = new Entity("Bird");
+    birdEntity.setProperty("birdName", request.getParameter("bird"));
```

Comment:
might want to check this isn't an empty string. otherwise, you might end up with a bunch of votes for bird "" in your database. also, might want to normalize bird names (e.g. to all caps, to all lower, or similar) in case someone hijacks your API and sends a bunch of bird names with different capitalization, giving different entries for essentially the same bird name.